### ISSUE TITLE
join() fails with an error on quantized models

### ISSUE THREAD

**Issue by hybridherbst** [donmccurdy/glTF-Transform#844]
**Describe the bug**
I think the new `optimize` command is designed as kind of a one-stop-shop for optimizing models; currently it seems to abort when any of the steps fails / can't do something:

```
PS C:\Users\herbst\Recordings> gltf-transform optimize .\Group-01.etc1s+meshopt.glb .\Group-01.etc1s+meshopt+opt.glb
warn: Decoded EXT_meshopt_compression. Further compression will be lossy.
✔ dedup                205ms
warn: instance: Instancing is not currently supported for animated models.
✔ instance             1ms
✔ flatten              9ms
✖ join: Requires ≥2 Primitives, sharing the same Material and Mode, with compatible vertex attributes and indices.

error: Requires ≥2 Primitives, sharing the same Material and Mode, with compatible vertex attributes and indices.
```

The file where this happens is confidential but I can send it via link somewhere if needed.

I believe there may be two issues here:
- `join` produces an error, not a warning, when it can't join something. I think it should be a warning instead.
- `optimize` stops doing something on the first error (assumption based on the above, haven't looked at the code)

**Expected behavior**
`optimize` always completes (unless there's really an error in the file)

**Versions:**
 - Version: 3.0.4
 - Environment: CLI

**Comment by donmccurdy** [comment]
The error reported here is (despite what the wording might suggest) an error and not a warning; it shouldn't be encountered in normal operation of the `join()` function, regardless of the input file. So I think ignoring the warning wouldn't be the right fix, there's a bug hidden underneath that.

If you can share an example that triggers the error I think we can get that fixed.

**Comment by hybridherbst** [comment]
Understood! After looking into it a bit more it looks like it _works_ for an original file and _fails_ for a file that has already been compressed with meshopt. I sent you the files on the glTF Slack.

**Comment by donmccurdy** [comment]
@hybridherbst I've pushed a partial fix (9731ae7bb93e7014c9ce685e98804e293977e9f5) that resolves the error when running `optimize` on the CLI. What is happening is that during the join() operation, glTF Transform may need to de-quantize mesh primitives in order to transform them into the same local space. It was doing that primitive-by-primitive, and would then hit a case where primitives it had already grouped for joining were no longer "compatible" after some were dequantized. So it was a real error that glTF Transform did not anticipate and couldn't recover from.

The short-term fix is just to run dequantize() before running join(), which I've added to the optimize command. You'll probably want to include `--compress meshopt` in order to reapply that compression at the end.

I need to think a little about the longer-term fix, since this would affect the join() function and `join` CLI command as well, which can be used outside of the `optimize` command. I haven't fixed those yet.
